consoles/video_stream.pm

```diff
@@ -225,6 +228,28 @@ sub send_key_event ($self, $key, $press_release_delay) {
       or die "failed to send '$key' input event";
 }
 
+sub type_string ($self, $args) {
+    my $keyboard_device_url = $bmwqemu::vars{GENERAL_HW_PICO_KEYBOARD_URL};
+    if ($keyboard_device_url) {
+        # send keyboard events using RPi Pico W based keyboard emulator
+        my $url = Mojo::URL->new($keyboard_device_url)->query(type => $args->{text});
+        bmwqemu::diag(Mojo::UserAgent->new->get($url)->result->body);
```

Comment:
Don't instantiate a new `Mojo::UserAgent` for every request (especially since this function expected to be called quite frequently). You could use e.g. `($self->{_ua} //= Mojo::UserAgent->new)` instead or maybe there's even already a ua instantiated somewhere in that context.